tsfc/kernel_interface/firedrake.py

```diff
@@ -120,13 +120,14 @@ def create_element(self, element, **kwargs):
 class KernelBuilder(KernelBuilderBase, KernelBuilderMixin):
     """Helper class for building a :class:`Kernel` object."""
 
-    def __init__(self, integral_data_info, scalar_type,
+    def __init__(self, integral_data_info, scalar_type, fem_scalar_type,
                  dont_split=(), diagonal=False):
         """Initialise a kernel builder."""
         integral_type = integral_data_info.integral_type
         subdomain_id = integral_data_info.subdomain_id
         domain_number = integral_data_info.domain_number
         super(KernelBuilder, self).__init__(scalar_type, integral_type.startswith("interior_facet"))
+        self.fem_scalar_type = fem_scalar_type
```

Comment:
Refactor so that `parameters["scalar_type"]` is the only thing that is passed in and the coffee builder constructs the C-name of the type internally.

Reply:
Then we don't need to pass two scalar types that are generally the same here